src/devices/Rtc/Devices/Ds3231/Ds3231Alarm2.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Iot.Device.Rtc
+{
+    /// <summary>
+    /// Represents alarm 2 on the DS3231
+    /// </summary>
+    public class Ds3231Alarm2
```

Comment:
The settings aren't the same. Alarm 1 is comprised of a day, hour, minute, second and match mode. Alarm 2 is comprised of a day, hour, minute and match mode. The match modes for each alarm are also different.

Reply:
I think the user should most of the time not have to worry about that. Throw an exception if one attempts to do something unsupported (ie. use alarm 2 with seconds), but otherwise keep these differences internal. You could change these classes to internal and use a public, common one in the public API.

Reply:
(internally you can have two classes but as public API I agree one would be much easier to use)

Reply:
If you feel they're distinct enough though please make sure to rename them accordingly, 1,2 suffix is not distinct enough and suggests similarity